amd/src/login.js

```diff
@@ -0,0 +1,214 @@
+        if ($remFromUserForm.prop('checked')) {
+            $remFromPassForm.val("1");
+        }
+    }
+
+    function setReturnurl($options) {
+        var retFromUserFormVal = $options.username.returnurl.val();
+        var $retFromPassForm = $options.password.returnurl;
+        if (retFromUserFormVal) {
+            $retFromPassForm.val(retFromUserFormVal);
+        }
+    }
+
+    function checkDomain(username, $options) {
+        var result = false;
+        queryWebservice(username)[0].done(function(response) {
+            result = handleDomainRedirect(response, username, $options);
+        }).fail(function() {
+            get_string_and_notify('danger', 'invalid_user');
+            result = false;
+        });
+        return result;
+    }
+
+    function queryWebservice(username){
+        return ajax.call([
+            {
+                methodname: WEB_SERVICE_METHOD_NAME,
+                args: {
+                    username: username
+                }
+            }
+        ]);
+    }
+
+    function handleDomainRedirect(response, username, $options) {
+        var currentURL = new URL(M.cfg.wwwroot);
+        if (response.email === null) {
+            // flag non-existent user error
+            get_string_and_notify('warning', 'non_existent_user');
+            return false;
+        } else if (response.domain !== currentURL.hostname) {
+            // Redirect to recorded user default domain
+            window.location = window.location.protocol + '//' + response.domain + currentURL.pathname +
+                '/local/signin/index.php?username=' + username;
+            return false;
+        } else {
+            $(defaults.form.alert).alert('close');
+            // Stay on page & flip over to password form
+            // Populate username field on password form
+            $options.password.username.val(username);
+            // Toggle the password form
+            $options.password.changeuser.click();
+            $options.password.submit.on('click', $options.dom, $options.password.validation);
+        }
+        return true;
+    }
+
+    /**
+     *
+     * @param type - ('Success', 'info', 'warning', 'danger')
+     * @param message - Message to be displayed
+     */
+    function notify(type, message) {
+        $(defaults.form.alert).alert('close');
+        $('<div>')
+            .addClass('alert alert-' + type + ' alert-dismissable')
+            .text(message)
+            .appendTo(defaults.form.window)
+    }
+
+    function toggleForms($event) {
+        $event.preventDefault();
+        var $options = $event.data;
+        $options.password.container.toggleClass('hide');
+        $options.username.container.toggleClass('hide');
+    }
+
+    /**
+     *
+     * @param notification_type - ('Success', 'info', 'warning', 'danger')
+     * @param string_ref - Pointer for lang/.../local_signin.php string
+     */
+    function get_string_and_notify(notification_type, string_ref){
```

Comment:
This name is really inconsistent with the naming of other methods. Can we use `camelCase` over `snake_case`?